R/ncdf.R

```diff
@@ -326,12 +335,21 @@ read_ncdf = function(.x, ..., var = NULL, ncsub = NULL, curvilinear = character(
       return(st_crs(NULL))
     }
   } else {
-    ret <- try(st_crs(ncmeta::nc_gm_to_prj(nc_grid_mapping)))
-    if(inherits(ret, "try-error")) {
-      return(st_crs(NULL))
-    } else {
-      return(ret)
-    }
+  	tryCatch({
+  		cv_units <- .get_gm_units(atts, rep_coord_var)
+  	
+  		base_gm <- ncmeta::nc_gm_to_prj(nc_grid_mapping)
+  	
+  		base_gm <- paste0(gsub("\\+units=m ", "", base_gm), 
+  						  " +units=", cv_units)
+  	
+    	st_crs(base_gm)
+  	}, error = function(e) {
+  		warning(paste0("failed to create crs based on grid mapping\n",
+  					  "and coordinate variable units. Will return NULL crs.\n",
+  					  "Original error: \n", e))
```

Comment:
This warning is hit by the Euro Cordex text when a lon/lat curvilinear dimension is created but there is also a grid mapping to try and use.